sympy/printing/tests/test_julia.py

```diff
@@ -212,7 +212,8 @@ def test_containers():
     assert julia_code(Tuple(*[1, 2, 3])) == "(1, 2, 3)"
     assert julia_code((1, x*y, (3, x**2))) == "(1, x.*y, (3, x.^2))"
     # scalar, matrix, empty matrix and empty list
-    assert julia_code((1, eye(3), Matrix(0, 0, []), [])) == "(1, [1 0 0;\n0 1 0;\n0 0 1], zeros(0, 0), Any[])"
+    from sympy.codegen.pynodes import List
```

Comment:
Should there be sympy/codegen/{julia,maple,octave}nodes.py files defining `List` or perhaps this means that `List` should be moved to a more general file in sympy/codegen. Obviously the import should be made top-level but I put it here for convenience of review.

Reply:
I'm not sure that List has the same meaning in all languages or if this should be considered an abstract "list" that would have different implementations in different languages.

Reply:
I think each code printer has its own way to print a `list, but since `sympify` is used on the input to `CodePrinter`s, this line causes the creation of a `Tuple`, which then has a `list` in it. Changing this to a `List` seems to avoid this problem, I'm not sure there is a better way. `List` is actually just a wrapper around `Tuple` which gets printed differently (if I understand correctly, it defaults to printing the same as `list`.

Reply:
I think you know more about this than I do so I can't really advise but I trust your judgement.

Reply:
@bjodah do you have any preference for how this is dealt with?

Reply:
I'm fine with moving it. I put it in pynodes to signal equivalence with Python lists.